### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by segiddins**
This is a fake PR for the purposes of code review.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+language: ruby
+cache: bundler
+rvm:
+  - 2.0.0-p451
+
+branches:
+  only:
+    - master
+    - /.+-stable$/
```
Comment (segiddins) [inline .travis.yml]: newline

File: cocoapods-search.gemspec [hunk 2]
```diff
@@ -6,11 +6,13 @@ require 'cocoapods-search/gem_version.rb'
 Gem::Specification.new do |spec|
   spec.name          = 'cocoapods-search'
   spec.version       = CocoapodsSearch::VERSION
-  spec.authors       = ['Emma Koszinowski']
-  spec.email         = ['emkosz@gmail.com']
-  spec.description   = %q{A short description of cocoapods-search.}
-  spec.summary       = %q{A longer description of cocoapods-search.}
-  spec.homepage      = 'https://github.com/EXAMPLE/cocoapods-search'
+  spec.authors       = ['Eloy Durán','Fabio Pelosin','Emma Koszinowski']
```
Comment (segiddins) [inline cocoapods-search.gemspec]: space after `,`

File: lib/cocoapods-search/command/search.rb [hunk 3]
```diff
@@ -1,43 +1,97 @@
...
+        ].concat(super.reject { |option, _| option == '--silent' })
+      end 
 
       def initialize(argv)
-        @name = argv.shift_argument
+        @use_regex = argv.flag?('regex')
+        @full_text_search = argv.flag?('full')
+        @stats = argv.flag?('stats')
+        @supported_on_ios = argv.flag?('ios')
+        @supported_on_osx = argv.flag?('osx')
```
Comment (segiddins) [inline lib/cocoapods-search/command/search.rb]: also need a `watchos` flag

File: lib/cocoapods-search/command/search.rb [hunk 4]
```diff
@@ -1,43 +1,97 @@
...
+        @web = argv.flag?('web')
+        @query = argv.arguments! unless argv.arguments.empty?
```
Comment (segiddins) [inline lib/cocoapods-search/command/search.rb:36]: the `unless` is unnecessary?

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -5,41 +5,83 @@
...
+  target.pod_targets = pod_targets
+  target
+end
+
 #-----------------------------------------------------------------------------#
+
+SpecHelper::Fixture.fixture('banana-lib') # ensure it exists
+SpecHelper::Fixture.fixture('orange-framework')
+
+
```
Comment (segiddins) [inline spec/spec_helper.rb]: extraneous trailing newline

File: lib/cocoapods-search/command/search.rb (hunk 3 above)
Comment (kylef) [inline lib/cocoapods-search/command/search.rb]: #6

File: .travis.yml (hunk 1 above)
Comment (emkosz) [inline .travis.yml]: Where do I need, or need to get rid of a newline?

Comment (segiddins) [inline .travis.yml]: it's missing the trailing newline, which can lead to merge conflicts (and my text editor automatically dirtying the file)

File: lib/cocoapods-search/command/search.rb (hunk 4 above)
Comment (emkosz) [inline lib/cocoapods-search/command/search.rb:36]: The test "complains for wrong parameters" fails without the unless. 
See terminal output:

Pod::Command::Search
  Search
    ✓ registers it self (11 ms)
    ✓ runs with correct parameters (21 ms)
    - complains for wrong parameters [FAILED]
    ✓ searches for a pod with name matching the given query ignoring case (3 ms)

File: cocoapods-search.gemspec (hunk 2 above)
Comment (emkosz) [inline cocoapods-search.gemspec]: this goes for the email addresses as well?

File: lib/cocoapods-search/command/search.rb (hunk 4 above)
Comment (segiddins) [inline lib/cocoapods-search/command/search.rb:36]: Ah ok I see why now, we want the `query` to be nil. :+1:

**Comment by emkosz** [comment]
Fixes the issue https://github.com/CocoaPods/CocoaPods/issues/3397

**Pull request closed without merging**